### PR TITLE
feat: add C implementation for `math/base/assert/is-composite`

### LINKED ISSUE #1901: [RFC]: add C implementation for `@stdlib/math/base/assert/is-composite`
### Description

This RFC proposes adding native C implementation for `math/base/assert/is-composite`:
```javascript
bool stdlib_base_uint32_is_composite( const double x );
```

### Related Issues

N/A

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by performant23**
This commit if applied adds the C implementation to `@stdlib/math/base/assert/is-composite` package

This pull request:

-   adds native C implementation to `@stdlib/math/base/assert/is-composite`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #1901 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by performant23** [comment]
working on the C examples and benchmarks!

Edit: have fixed the .c file! Requesting you to review the PR. Thanks!

Note: The current function checks if a number is composite using:
```javascript
return ( stdlib_base_is_integer( x ) && ( x > 1 ) && ( x != STDLIB_CONSTANT_FLOAT64_PINF )  && ( !stdlib_base_is_prime( x ) ) );
```
This is written according to the function written in `lib/main.js`. Please do let me know if we are to prefer something like:
```javascript
...
if(condition1){
   return false;
}
...
```
(using multiple if statements for filtering instead of performing `AND` on the checks in one go). 
cc @Pranavchiku

File: lib/node_modules/@stdlib/math/base/assert/is-composite/benchmark/benchmark.native.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+// MAIN //
+
+bench( pkg+'::native', opts, function benchmark( b ) {
+	var bool;
+	var i;
+
+	b.tic();
+	for ( i = 0; i < b.iterations; i++ ) {
+		bool = isComposite( i + 1 );
+		if ( typeof bool !== 'boolean' ) {
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/assert/is-composite/benchmark/benchmark.native.js]:
```suggestion
		if ( !isBoolean( bool ) ) {
```

Comment (Pranavchiku) [reply lib/node_modules/@stdlib/math/base/assert/is-composite/benchmark/benchmark.native.js]: I'll add all suggestions at once, please wait

File: lib/node_modules/@stdlib/math/base/assert/is-composite/manifest.json [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      "libraries": [],
+      "libpath": [],
+      "dependencies": [
+        "@stdlib/math/base/assert/is-prime",
+        "@stdlib/math/base/assert/is-integer",
+        "@stdlib/constants/float64/pinf"
+      ]
+    }
+  ]
+}
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/assert/is-composite/manifest.json]:
```suggestion
{
  "options": {},
  "fields": [
    {
      "field": "src",
      "resolve": true,
      "relative": true
    },
    {
      "field": "include",
      "resolve": true,
      "relative": true
    },
    {
      "field": "libraries",
      "resolve": false,
      "relative": false
    },
    {
      "field": "libpath",
      "resolve": true,
      "relative": false
    }
  ],
  "confs": [
    {
      "src": [
        "./src/main.c"
      ],
      "include": [
        "./include"
      ],
      "libraries": [],
      "libpath": [],
      "dependencies": [
        "@stdlib/math/base/assert/is-prime",
        "@stdlib/math/base/assert/is-integer",
        "@stdlib/constants/float64/pinf"
      ]
    }
  ]
}
```

File: lib/node_modules/@stdlib/math/base/assert/is-composite/src/main.c [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/math/base/assert/is_composite.h"
+#include "stdlib/math/base/assert/is_prime.h"
+#include "stdlib/math/base/assert/is_integer.h"
+#include "stdlib/constants/float64/pinf.h"
+#include <stdint.h>
+#include <math.h>
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/assert/is-composite/src/main.c]:
```suggestion
```

I think we don't need these

File: lib/node_modules/@stdlib/math/base/assert/is-composite/test/test.native.js [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+var NINF = require( '@stdlib/constants/float64/ninf' );
+var tryRequire = require( '@stdlib/utils/try-require' );
+
+
+// FIXTURES //
+
+var PRIMES = require( './fixtures/primes.js' );
+
+
+// VARIABLES //
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/assert/is-composite/test/test.native.js:38]:
```suggestion
// VARIABLES //
```

**Review by Pranavchiku — APPROVED** [review]
Looks good. Thanks @performant23 !

File: lib/node_modules/@stdlib/math/base/assert/is-composite/lib/native.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+
+// MAIN //
+
+/**
+* Returns a boolean indicating whether a number is a composite.
+*
+* @private
+* @param {number} x - value to test
+* @returns {boolean} boolean indicating whether the number is composite.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/assert/is-composite/lib/native.js]:
```suggestion
* @returns {boolean} boolean indicating whether the number is composite
```

File: lib/node_modules/@stdlib/math/base/assert/is-composite/README.md [hunk 6]
```diff
@@ -80,6 +80,95 @@ bool = isComposite( NaN );
...
+```
+
+#### stdlib_base_is_composite( x )
+
+Tests if a finite double-precision floating-point number is a composite number.
+
+```c
+#include <stdbool.h>
+
+bool out = stdlib_base_is_composite( 3 );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/assert/is-composite/README.md]:
```suggestion
bool out = stdlib_base_is_composite( 3.0 );
```

**Review by Planeshifter — APPROVED** [review]

**Pull request merged**